Dockerfile

```diff
@@ -106,9 +117,18 @@ ARG DEVEL=no
 # i.e. 'docker compose run --rm web python -m warehouse shell --type=ipython')
 ARG IPYTHON=no
 
+# By default, Docker has special steps to avoid keeping APT caches in the layers, which
+# is good, but in our case, we're going to mount a special cache volume (kept between
+# builds), so we WANT the cache to persist.
+RUN set -eux; \
```

Comment:
Did you mean to do this twice? It looks like this modifies the state of the container so that it's *always* caching, so I suspect we don't need it a second time?

Reply:
once for docs target and the other for the build target.

Reply:
Since build starts `FROM python:3.11.4-slim-bullseye` we need to re do this. It re-uses the cache and adds to it.

When we `FROM python:3.11.4-slim-bullseye` for the final image, we once again just use the cache, but let it get obliterated.